### ISSUE TITLE
Gothic1 crashes witch unhandled exception while switching back to desktop since v17.7-dev30

### ISSUE THREAD

**Issue by stefson** [SaiyansKing/GD3D11#27]
so I initially noticed this when your efforts were backported/merged into kiridis renderer, so I went to work testing all available intermediate versions of your fork to hopefully find more helpfull information. 

But first of all, the actuall error:

```
======================================= UNHANDLED EXCEPTION OCCURED ======================================================
======================================= CRASH INFOS: =====================================================================
GOTHIC 1.08k_mod, Parser Version: 50
User:  tim,  CPUType: 586,  Mem: 2048 MB total, 2048 MB free
Startup Options:-game:dunkle_geheimnisse.ini
=============================================== CALLSTACK : ==============================================================
0023:00000000 (0x0BA98BEC 0x68A68ED0 0x00000000 0x0BA98B24) GothicMod.exe
0023:023794B9 (0x0BA98B24 0x1471C0E8 0x0126F8F4 0x0075FA11) g1_avx.dll
0023:0237A998 (0x0BA98B24 0x1471C168 0x1471C0E8 0x00000000) g1_avx.dll
0023:0075FA11 (0x00000001 0x00000000 0x1471C0E8 0x00000000) GothicMod.exe, CD3duContext::_DestroyDXBuffers()+174 byte(s)
0023:0075FCBC (0x01ED91DC 0x0075E82D 0x00000000 0x0075C98B) GothicMod.exe, CD3duContext::~CD3duContext()+182 byte(s)
0023:0075B702 (0x00000000 0x0075C98B 0x01ED91DC 0x0075E9A6) GothicMod.exe, CD3duContext::`scalar deleting destructor'()+8 byte(s)

0023:0075C98B (0x00000001 0x00000000 0x0075ECFD 0x1471C0E8) GothicMod.exe, CD3duContextNode::`scalar deleting destructor'()+8 byte(s)
0023:0075E9A6 (0x1471C0E8 0x007D06E8 0x00712828 0x1471C0E8) GothicMod.exe, CD3duContextList::DestroyD3duContext()+38 byte(s)
0023:0075ECFD (0x1471C0E8 0x7FD50034 0x00000000 0x0126FA48) GothicMod.exe, CD3duContext::Release()+29 byte(s)
0023:00712828 (0x00120000 0x00000000 0x7FD50034 0x7BC26BEE) GothicMod.exe, zCRnd_D3D::XD3D_ClearDevice()+1080 byte(s), D:\dev\gothic\current_work\ZenGin\_Kurt\zRndD3D_Init.cpp, line 1168
0023:7BC26BEE (0x00120094 0x01ED00B0 0x0BA98FD4 0x14664F34) ntdll.dll, RtlIsValidIndexHandle()+8318 byte(s)
0023:7BC26BEE (0x00000000 0x00000690 0x0000041A 0x00000020) ntdll.dll, RtlIsValidIndexHandle()+8318 byte(s)
0023:00000000 (0x00717150 0x00716F50 0x00714B60 0x00716860) GothicMod.exe
```

I tested all your builds from v17.7-dev21 on and ended up with v17.7-dev29 being the last clean one, v17.7-dev30 is the first version with the bug. The diff is very small: https://github.com/SaiyansKing/GD3D11/compare/v17.7-dev29...v17.7-dev30 

I also keep seeing many of these errors most likely linked to the msvc compiler
```
0114:err:msvcrt:symbol_demangle Unknown operator: i
0114:err:msvcrt:demangle_datatype Unknown type @
```
did you maybe change anything in your compiler setup that isn't documented in the diff of the source code? 

find attached two logs from gothic1/system: [logs-from-system.zip](https://github.com/SaiyansKing/GD3D11/files/7734455/logs-from-system.zip)

I'm using wine to play gothic1 dx11 render, through dxvk. Wine versions is latest 7.0_rc1

**Comment by stefson** [comment]
crash is fixed for me with https://github.com/SaiyansKing/GD3D11/commit/ae762dc5c743d3314e29afd1feac9a07e418c329 , part of v17.8-dev2-fix2

**Comment by SaiyansKing** [comment]
Hi,

> I also keep seeing many of these errors most likely linked to the msvc compiler
> 
> ```
> 0114:err:msvcrt:symbol_demangle Unknown operator: i
> 0114:err:msvcrt:demangle_datatype Unknown type @
> ```
> 
> did you maybe change anything in your compiler setup that isn't documented in the diff of the source code?

no I didn't change any project settings if that what you mean but I remember updating platform toolsets.

About your keyboard issue I don't know what could cause that because all keyboard inputs goes through different library that is called Direct Input also Zenigne force keyboard inputs to be always acquired so even with settings open by F11 it doesn't disable keyboard.

**Comment by stefson** [comment]
Yes, that keyboard bug is very strange. I will do more research in the next days, the issue here is fixed with 17.8-dev2-fix2
